Client/Frontend/Browser/LocalRequestHelper.swift

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+
+class LocalRequestHelper: BrowserHelper {
+    func scriptMessageHandlerName() -> String? {
+        return "localRequestHelper"
+    }
+
+    func userContentController(userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        guard message.frameInfo.request.URL?.isLocal ?? false else { return }
+
+        let params = message.body as! [String: String]
+
+        if params["type"] == "load",
+           let url = NSURL(string: params["url"]!) {
```

Comment:
> we cannot trust data coming in via JS

In this case, we can -- we own all of the JS sending these messages (guaranteed by the`isLocal` `guard` here), so any bad message is a programmer error.

That said, I have an `assertionFailure()` at the end of this `if` chain already, so I'm fine switching it to optional since we'll instead fail the assertion if this ever happens.